Content.Shared/Clothing/EntitySystems/ClothingSystem.cs

```diff
@@ -176,6 +174,15 @@ private void OnUnequipDoAfter(Entity<ClothingComponent> ent, ref ClothingUnequip
             _handsSystem.TryPickup(args.User, ent);
     }
 
+    private void CheckEquipmentForLayerHide(string slot, EntityUid equipment, EntityUid equipee)
+    {
+        if ((new string[] { "head" }).Contains(slot) && _tagSystem.HasTag(equipment, HairTag))
+            ToggleVisualLayer(equipee, HumanoidVisualLayers.Hair, HairTag);
+        if ((new string[] { "mask", "head" }).Contains(slot) && _tagSystem.HasTag(equipment, NoseTag))
+            if (TryComp(equipee, out HumanoidAppearanceComponent? appearance) && appearance.HideNoseOnEquip)
+                ToggleVisualLayer(equipee, HumanoidVisualLayers.Snout, NoseTag);
+    }
+
```

Comment:
This isn't your fault but this code is extremely ass:
- it should be using the enums not strings.
- It shouldn't be re-allocating a string array every single time this is called.
- It shouldn't be hardcoded to specific layers and copy-pasted, it should just be generic between the 2 i.e. "HideNoseOnEquip" should be a bitmask of the slots.
- TagComponent should be re-used between them.